pkg/puller/watcher.go

```diff
@@ -0,0 +1,126 @@
+package puller
+
+import (
+	"encoding/json"
+	"github.com/fsnotify/fsnotify"
```

Comment:
file watchers are notorious for being brittle. I'm a little nervous about relying on this.

Reply:
Viper relies on that as well. But tbh, I’d prefer to have containers look at file-system and be k8s agnostic. Relying on api-server doesn’t sound necessarily better.

Reply:
Also, where do you see file watChers being brittle? I purposefully decided to use it instead of configmap informer. My local testing produced good results